### PR TITLE
Move model loading out of request handler

### PR DESCRIPTION
**Pull request by rhohndorf**

### CODE REVIEW

File: update_validator.sh [hunk 1]
```diff
@@ -28,6 +32,47 @@ source_script "update_env.sh"
...
+        fi
+        (( attempt++ ))
+    done
+
+    echo "Error: Failed to restart $process_name after $max_attempts attempts."
+    return 1
+}
+
+# Restart validation process up to 5 times
+restart_with_retry "$VALIDATION_PROCESS" 5
```
Comment (rhohndorf) [inline update_validator.sh:75]: This fails with error
[PM2][ERROR] Process or Namespace validation not found
if the validation process was not started prior to running the update script. Is this the intended behavior? Would it be better to check if it's running and then either start or restart?

File: update_validator.sh [hunk 2]
```diff
@@ -28,6 +32,47 @@ source_script "update_env.sh"
...
+
+# Restart validator process
+pm2 restart "$VALIDATOR_PROCESS"
```
Comment (rhohndorf) [inline update_validator.sh:78]: see above

File: update_validator.sh (hunk 1 above)
Comment (the-mx) [reply update_validator.sh:75]: it's expected to be running.

File: update_validator.sh (hunk 2 above)
Comment (the-mx) [reply update_validator.sh:78]: Same

**Pull request merged**